src/coreComponents/dataRepository/BufferOpsDevice.cpp

```diff
@@ -145,25 +145,25 @@ PackDataByIndexDevice ( buffer_unit_type * & buffer,
                         parallelDeviceEvents & events )
 {
   localIndex const numIndices = indices.size();
-  localIndex const unitSize = var.size() / var.size( 0 ) * sizeof( T );
-
-  buffer_unit_type * const devBuffer = buffer;
+  localIndex const sliceSize = var.size() / var.size( 0 );
+  localIndex packedSize = numIndices * sliceSize * sizeof( T );
   if( DO_PACKING )
   {
+    T * devBuffer = reinterpret_cast< T * >( buffer );
     parallelDeviceStream stream;
```

Comment:
We should clean up this stream and events business later. We should not be creating a new stream for every field that we pack and we should probably do stream based synchronization instead of event based.